bot.go

```diff
@@ -36,7 +37,9 @@ func NewBot(f ...func(*Bot)) *Bot {
 		Data:      make(chan *irc.Message),
 		callbacks: make(map[string][]Callback),
 		tries:     0,
+		stop:      make(chan bool),
 	}
+	b.stop <- false
```

Comment:
This would look something like this:

`func (b *Bot) Disconnect() {
    close(b.stop)
}```

Every listener on this channel would receive, and we could close the read loop and the callback loop. Then, we could call `Connect()` again, and re-create those loops.